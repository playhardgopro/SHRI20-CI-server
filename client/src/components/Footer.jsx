import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import './scss/Footer.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Footer = ({ children, className }) => {
  return (
    <div className="footer">
      <div className="footer__content">
        <div className="footer__links">
          <a href="#">
            <div className="text text_view_ghost text_size_13-18 text_type_link">Support</div>
          </a>
          <a href="#">
            <div className="text text_view_ghost text_size_13-18 text_type_link">Learning</div>
          </a>
        </div>
        <div className="footer__copyright">
          <div className="text text text_size_13-18 text_view_ghost">&copy; 2020 Your Name</div>
        </div>
      </div>
    </div>
  )
}

Footer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf([PropTypes.string, PropTypes.bool]),
}
Footer.defaultProps = {
  children: '',
  className: {},
}

export default Footer
