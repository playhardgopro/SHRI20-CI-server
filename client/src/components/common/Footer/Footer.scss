@import '../../var.scss';
$screen-m: 768px;
.footer {
  flex-shrink: 0;
  background-color: var(--blue-grey-50);
}

.footer__links a {
  margin-right: var(--space-12);
}

.footer__content {
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
  max-width: 1024px;
  padding: var(--space-7) var(--padding);
}

.footer__links {
  display: flex;
  align-items: center;
}

.footer__copyright {
  display: flex;
  align-items: center;
}

@media screen and (max-width: $screen-m) {
  .footer__content {
    flex-direction: column;
    padding-top: var(--space-12);
    padding-bottom: var(--space-12);
  }

  .footer__content > * {
    margin-bottom: var(--space-8);
  }
}
