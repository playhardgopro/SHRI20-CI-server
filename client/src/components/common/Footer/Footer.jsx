import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { withNaming } from '@bem-react/classname'
import './Footer.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Footer = ({ children, className }) => {
  const cnFooter = cn('footer')
  const cnText = cn('text')
  return (
    <div className={cnFooter()}>
      <div className={cnFooter('content')}>
        <div className={cnFooter('links')}>
          <Link to="#">
            <div className={cnText({ size: '13-18', view: 'ghost', type: 'link' })}>Support</div>
          </Link>
          <Link to="#">
            <div className={cnText({ size: '13-18', view: 'ghost', type: 'link' })}>Learning</div>
          </Link>
        </div>
        <div className={cnFooter('copyright')}>
          <div className={cnText({ size: '13-18', view: 'ghost' })}>&copy; 2020 Daniil Egortsev</div>
        </div>
      </div>
    </div>
  )
}

Footer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Footer.defaultProps = {
  children: '',
  className: {},
}

export default Footer
