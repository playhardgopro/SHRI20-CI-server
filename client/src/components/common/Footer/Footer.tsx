import * as React from 'react'
import { Link } from 'react-router-dom'
import { withNaming } from '@bem-react/classname'
import './Footer.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Footer = () => {
  const cnFooter = cn('footer')
  const cnText = cn('text')
  return (
    <div className={cnFooter()}>
      <div className={cnFooter('content')}>
        <div className={cnFooter('links')}>
          <Link to="#">
            <div className={cnText({ size: '13-18', view: 'ghost', type: 'link' })}>Support</div>
          </Link>
          <Link to="#">
            <div className={cnText({ size: '13-18', view: 'ghost', type: 'link' })}>Learning</div>
          </Link>
        </div>
        <div className={cnFooter('copyright')}>
          <a href="https://github.com/playhardgopro">
            <div className={cnText({ size: '13-18', view: 'ghost', type: 'link' })}>&copy; 2020 Daniil Egortsev</div>
          </a>
        </div>
      </div>
    </div>
  )
}

export default Footer
