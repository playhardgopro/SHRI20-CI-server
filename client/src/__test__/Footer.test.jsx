import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../components/common/Footer/Footer'

jest.mock('react-router-dom', () => ({
  Link: (props) => <a href={props.to}>{props.children}</a>,
}))

it('renders Footer correctly', () => {
  const footer = renderer.create(<Footer />).toJSON()

  expect(footer).toMatchSnapshot()
})
