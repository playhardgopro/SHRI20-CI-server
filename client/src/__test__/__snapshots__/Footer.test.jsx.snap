// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="footer__content"
  >
    <div
      className="footer__links"
    >
      <a
        href="#"
      >
        <div
          className="text text_size_13-18 text_view_ghost text_type_link"
        >
          Support
        </div>
      </a>
      <a
        href="#"
      >
        <div
          className="text text_size_13-18 text_view_ghost text_type_link"
        >
          Learning
        </div>
      </a>
    </div>
    <div
      className="footer__copyright"
    >
      <a
        href="https://github.com/playhardgopro"
      >
        <div
          className="text text_size_13-18 text_view_ghost text_type_link"
        >
          Â© 2020 Daniil Egortsev
        </div>
      </a>
    </div>
  </div>
</div>
`;
